src/components/TodoList.tsx

```diff
@@ -0,0 +1,295 @@
+import classNames from 'classnames';
+import {
+  FC, FormEvent, useContext, useRef, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodo } from '../api/todos';
+import { AuthContext } from './Auth/AuthContext';
+import { Loader } from './Loader';
+import { ErrorContext } from './ErrorContext';
+
+type Props = {
+  visibleTodos: Todo[],
+  isNewTodoLoaded: boolean,
+  setVisibleTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setClickedIndex: React.Dispatch<React.SetStateAction<number>>,
+  clickedIndex: number,
+  isCompletedTodosDeleting: boolean,
+  areTodosToggling: boolean,
+};
+
+export const TodoList: FC<Props> = ({
+  isNewTodoLoaded,
+  visibleTodos,
+  setVisibleTodos,
+  clickedIndex,
+  setClickedIndex,
+  isCompletedTodosDeleting,
+  areTodosToggling,
+}) => {
+  const user = useContext(AuthContext);
+  const [isTodoDeleted, setIsTodoDeleted] = useState(true);
+  const [isTodoToggled, setIsTodoToggled] = useState(false);
+  const [isTodoEdited, setIsTodoEdited] = useState(true);
+  const [isTodoEditing, setIsTodoEditing] = useState(false);
+
+  const {
+    setIsRemoveErrorShown,
+    setHasLoadingError,
+    setIsEmptyTitleErrorShown,
+    setIsTogglingErrorShown,
+    setIsAddingErrorShown,
+  } = useContext(ErrorContext);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [inputValue, setInputValue] = useState('');
+
+  function setErrorsToFalseExceptRemoveError() {
+    setIsEmptyTitleErrorShown(false);
+    setHasLoadingError(false);
+    setIsTogglingErrorShown(false);
+    setIsAddingErrorShown(false);
+
+    setIsRemoveErrorShown(true);
+  }
+
+  function setErrorsToFalseExceptTogglingError() {
+    setIsEmptyTitleErrorShown(false);
+    setHasLoadingError(false);
+    setIsRemoveErrorShown(false);
+    setIsAddingErrorShown(false);
+
+    setIsTogglingErrorShown(true);
+  }
+
+  const deleteTodoHandler = (todoId: number) => {
+    setIsTodoDeleted(false);
+
+    if (user) {
+      deleteTodo(user.id, todoId)
+        .then((() => {
+          setIsTodoDeleted(true);
+          setVisibleTodos(prevTodos => prevTodos.filter(x => x.id !== todoId));
+          setIsRemoveErrorShown(false);
+        }))
+        .catch(() => {
+          setErrorsToFalseExceptRemoveError();
+          setIsTodoDeleted(true);
+        });
```

Comment:
It would be good to decide on the method of asynchronous requests and use one thing in the whole project (async/await or then).